sdk/nodejs/resource.ts

```diff
@@ -269,6 +281,10 @@ export abstract class Resource {
             this.__providers = { ...this.__providers, ...providers };
         }
 
+        // Combine transformations inherited from the parent with transformations provided in opts.
+        const parent = opts.parent || getStackResource() || { __transformations: undefined };
```

Comment:
wait... can `getStackResource` return null?

Reply:
got is now.  this is sxs.  please doc as such.

Reply:
Yes - though only very early in the initialization.  It's possible some other invariant should ensure this can't happen by the point this code runs - but better to not crash here in case it does.